### ISSUE TITLE
Incomplete aria-expanded in query_sort_button_component.html.erb

### ISSUE THREAD

**Issue by metamoni** [joemasilotti/railsdevs.com#539]
## Describe the bug

The "Sort" button on the Developers list page uses `aria-expanded="false"` to tell screen readers when the sort menu is collapsed, but is not set up to change to `true` when the menu is expanded.

File: `query_sort_button_component.html.erb`

## To reproduce (on a Mac)

1. Sign in as a developer
2. Go to http://localhost:3000/developers
3. Turn on VoiceOver: `command + F5`
4. Tab to "Sort" and notice the screen reader announces the menu is "collapsed"
5. Hit `Enter` to open the menu and notice the screen reader announces "menu (2 items)"
6. Turn VoiceOver off: `command + F5`

## Expected behavior
When opening the "Sort", the screen reader should announce that it's "expanded"
When closing the "Sort" menu, the screen reader should announce that it's "collapsed"

See example of correct screen reader behaviour below:

https://user-images.githubusercontent.com/22390758/177088249-951fce1a-7786-41af-abc6-b67fc7462feb.mov

## Actual behavior
Expanded/collapsed state isn't announced when the "Sort" menu is open

Current screen reader behavior

https://user-images.githubusercontent.com/22390758/177088192-59640620-7c5d-4acb-8c1e-0f66c0cde8d4.mov

## Additional context
Example of using `aria-expanded` properly with an accordion component: [w3c.github.io](https://w3c.github.io/aria-practices/examples/accordion/accordion.html)

**Comment by axhgl** [comment]
Could I give this one a try, @joemasilotti? Thank you @metamoni for a well described issue along with the additional context.

**Comment by joemasilotti** [comment]
Sure thing - all yours!

**Comment by joemasilotti** [comment]
Closed via #585.
